src/ophyd_async/core/detector.py

```diff
@@ -307,11 +336,9 @@ async def describe_collect(self) -> Dict[str, Descriptor]:
     async def collect_asset_docs(
         self, index: Optional[int] = None
     ) -> AsyncIterator[StreamAsset]:
-        """Collect stream datum documents for all indices written.
-
-        The index is optional, and provided for flyscans, however this needs to be
-        retrieved for stepscans.
-        """
+        # Collect stream datum documents for all indices written.
```

Comment:
The protocol provides an identical docstring, so its not worth repeating. Usually not worth bothering with docstrings for override methods because it's just pointless duplication. Your IDE will work it out.